core/src/HIP/Kokkos_HIP_Instance.hpp

```diff
@@ -61,6 +61,7 @@ struct HIPTraits {
       1024;  // FIXME_HIP -- assumed constant for now
 
   static int constexpr ConstantMemoryUsage        = 0x008000; /* 32k bytes */
+  static int constexpr KernelArgumentLimit        = 0x001000; /*  4k bytes */
```

Comment:
I'm really not a big fan of putting an unreadable number in the code and putting the readable version in the comment. Why not just say `KernelArgumentLimit = 4096; /* bytes */`?